.github/scripts/sync_labels.py

```diff
@@ -48,12 +48,13 @@ def get_linked_pr_from_issue_number(repo, number):
 
 def get_linked_issues_based_on_pr_body(repo, number):
     pr = repo.get_pull(number)
-    pattern = fr'Fixes:? (?:#|{repo}#|https://github.com/{repo}/issues/)(\d+)'
-    matches = re.findall(pattern, pr.body)
+    pattern = fr'(fix(?:|es|ed)|resolve(?:|d|s)|close(?:|s|d))?(|{repo}|https.*?/issues/$)(\d+)'
```

Comment:
the trailing `$` in the second group never matches. as it indicates the end of a line.

and why did you change `https://github.com/{repo}` to `https.*?` ? just to be more tolerant? 

and if we are using non-capturing match group, why not apply it to all of them?

if we are using `?` to indicate an optional match of the first matching group, why not apply it to the second one? 

in short, i'd suggest:
```py
pattern = fr'(?:fix(?:|es|ed)|resolve(?:|d|s)|close(?:|s|d))?(?:{repo}|https://github.com/{repo}/issues/)?(\d+)'
```

so that we don't need to have
```py
match = next((num for num in match if num), None)
```
as `matches` is a list of `str` instead of a list of tuple of `str` with the proposed change.

Reply:
@tchaikov applied your suggestion with a minor change removing the `f` from the pattern, otherwise i can't remove `match = next((num for num in match if num), None)`

Reply:
i don't get this one? why? it should be addressed by the leading `?:` in all capturing groups, right?

Reply:
Well, the result I am getting for the matching pattern is `('', '', '59')` rather than `59`. So i am not sure what am i missing here

Reply:
weird.. i did test the proposed regex locally. only a single matched group is returned.